osbuild/mounts.py

```diff
@@ -58,7 +58,21 @@ def __init__(self, devices: DeviceManager, root: str) -> None:
 
     def mount(self, mount: Mount) -> Dict:
 
-        source = self.devices.device_abspath(mount.device)
+        # Get the absolute path to the source device inside the
+        # temporary filesystem (i.e. /run/osbuild/osbuild-dev-xyz/loop0)
+        # and also the relative path to the source device inside
+        # that filesystem (i.e. loop0). If the device also exists on the
+        # host in `/dev` (like /dev/loop0), we'll use that path for the
+        # mount because some tools (like grub2-install) consult mountinfo
+        # to try to canonicalize paths for mounts and inside the bwrap env
+        # the device will be under `/dev`. https://github.com/osbuild/osbuild/issues/1492
+        source = None
```

Comment:
Would it be possible to write a small unit test for this with some example inputs? Or too is that too annyoing as too much stuff would have to get mocked? When reading this I was thinking that some examples in a test would help (me at least) to get a better idea what is going on

Reply:
I'm not sure how easy it would be to write a test for this @achilleas-k might know. 

Would it go in `test_mount.py`?

Reply:
@mvo5 since this is blocking a few other things maybe we could get it in and then you and I can work together after the holidays to write a test for it? I think a quick guided session by you would help me understand a little better what you are looking for from the test and where you think the best place to put it would be.

Reply:
Sure, let's do that